src/cron/exportToTinybird.ts

```diff
@@ -24,64 +53,144 @@ export const getPostsTinybirdExport = (con: DataSource, latest: Date) =>
          and "sourceId" != '${UNKNOWN_SOURCE}'
          and "visible" = true
          and "type" != '${PostType.Welcome}'
+       ORDER BY "id"
       `,
-    [latest],
-  );
+      [latest],
+    );
+  }
+}
+export interface IPostsMetadataRepository {
+  latest(): Promise<Date>;
+  append(posts: TinybirdPost[]): Promise<PostDatasourceResult>;
+}
+
+export class PostsMetadataRepository implements IPostsMetadataRepository {
+  private readonly tinybirdClient: ITinybirdClient;
+  private readonly datasource: string;
+  private readonly latestQuery: string;
+  constructor(tinybirdClient: ITinybirdClient, datasource: string) {
+    this.tinybirdClient = tinybirdClient;
+    this.datasource = datasource;
+
+    this.latestQuery = `
+      SELECT
+          max(metadata_changed_at) as latest
+      FROM ${datasource}
+      FORMAT JSON
+    `;
+  }
+
+  public async latest(): Promise<Date> {
+    interface latest {
+      latest: string;
+    }
+
+    const result = await this.tinybirdClient.query<latest>(this.latestQuery);
+    if (result.rows === 0) {
+      throw new Error('no rows returned');
+    }
+
+    return new Date(result.data[0].latest);
+  }
+
+  public async append(posts: TinybirdPost[]): Promise<PostDatasourceResult> {
+    const csv: string = await TinybirdClient.Json2Csv(posts);
+
+    return await this.tinybirdClient.postToDatasource(
+      this.datasource,
+      TinybirdDatasourceMode.APPEND,
+      csv,
+    );
+  }
+}
+export interface TinybirdExportResult {
+  exported: number;
+  since: Date;
+  tinybird: PostDatasourceResult;
+}
+export class TinybirdExportService {
+  private readonly logger: FastifyBaseLogger;
+  private readonly postsRepository: IPostsRepository;
+  private readonly postsMetadataRepository: IPostsMetadataRepository;
+
+  constructor(
+    logger: FastifyBaseLogger,
+    postsRepository: IPostsRepository,
+    postsMetadataRepository: IPostsMetadataRepository,
+  ) {
+    this.logger = logger;
+    this.postsRepository = postsRepository;
+    this.postsMetadataRepository = postsMetadataRepository;
+  }
+
+  public async export(): Promise<TinybirdExportResult> {
+    const latest = await this.postsMetadataRepository.latest();
+    const postsToExport = await this.postsRepository.getForTinybirdExport(
+      latest,
+    );
 
-const jsonexportPromise = promisify(jsonexport);
+    if (postsToExport.length === 0) {
+      return {
+        since: latest,
+        exported: 0,
+        tinybird: null,
+      };
+    }
+
+    const tinybirdResult = await this.postsMetadataRepository.append(
+      postsToExport,
+    );
+
+    return {
+      since: latest,
+      exported: postsToExport.length,
+      tinybird: tinybirdResult,
+    };
+  }
+
+  public async exportAndLog(): Promise<void> {
+    let result: TinybirdExportResult;
+
+    try {
+      result = await this.export();
+    } catch (err) {
+      this.logger.error(
+        { error: err.message, stack: err.stack },
```

Comment:
We usually log errors like this

```suggestion
        { err },
```